### PR TITLE
Enhance button theme to support icon-only padding specification

### PR DESCRIPTION
**Pull request by taysea**
#### What does this PR do?

Adds support for custom padding for icon-only buttons (buttons that have `icon` and no `label`). Having the ability to specify icon only padding is desirable to create uniform dimension icon only buttons.

Additionally, it is desirable to be able to manage icon size from the theme and vary the icon's size depending on what size button it is contained in.

This PR adds:

- theme.button.icon.size (allows caller to specify what size the icon should render as when contained in a small, medium, or large button
```
  icon?: {
    size?: {
      small?: string;
      medium?: string;
      large?: string;
    };
  };
```

- theme.button.size[size].iconOnly.pad which allows the caller to specify pad for iconOnly pad across button sizes.

#### Where should the reviewer start?

base.js

#### What testing has been done on this PR?

Tested in local story:

Screen Shot 2023-01-26 at 2 44 10 PM

#### Do Jest tests follow these best practices?

- [ ] `screen` is used for querying.
- [ ] The correct query is used. (Refer to [this list of queries](https://testing-library.com/docs/queries/about/#priority))
- [ ] `userEvent` is used in place of `fireEvent`.
- [x] `asFragment()` is used for snapshot testing.

#### What are the relevant issues?

Related to: https://github.com/grommet/hpe-design-system/issues/3117

#### Do the grommet docs need to be updated?
Yes.

#### Should this PR be mentioned in the release notes?
Yes.

#### Is this change backwards compatible or is it a breaking change?
Backwards compatible.

### CODE REVIEW

**Review by ericsoderberghp — APPROVED** [review]

File: src/js/components/Button/StyledButtonKind.js [hunk 1]
```diff
@@ -37,7 +37,16 @@ const fontStyle = (props) => {
...
 
-const padFromTheme = (size = 'medium', theme, themeObj, kind) => {
+const padFromTheme = (size = 'medium', theme, themeObj, kind, iconOnly) => {
+  if (size && iconOnly && themeObj?.size?.[size]?.iconOnly?.pad) {
+    const pad = themeObj?.size?.[size]?.iconOnly?.pad;
+
+    return {
+      vertical: typeof pad === 'string' ? pad : pad.vertical,
+      horizontal: typeof pad === 'string' ? pad : pad.horizontal,
+    };
```
Comment (halocline) [inline src/js/components/Button/StyledButtonKind.js:47]: Minor. Not sure if either of these helps readability... 

```suggestion
    return typeof pad === 'object' ? pad : {
      vertical: pad,
      horizontal: pad,
    };
```

OR 

```suggestion
    return {
      vertical: pad.vertical || pad,
      horizontal: pad.horizontal || pad,
    };
```

File: src/js/components/Button/stories/CustomThemed/Active.js [hunk 2]
```diff
@@ -1,26 +1,318 @@
...
-  >
-    <Box pad="large" align="center">
-      <Button hoverIndicator="light-1" onClick={() => {}} active>
-        {/*  When kind Button include children, it is treated as plain */}
-        <Box pad="small" direction="row" align="center" gap="small">
-          <Add />
-          <Text>Kind</Text>
-        </Box>
-      </Button>
+          height: '24px',
```
Comment (halocline) [inline src/js/components/Button/stories/CustomThemed/Active.js:144]: Noticing this comment and the height value. Is this working? I don't see the effect coming through. Also don't see it represented in base.d.ts or base.js.

Comment (taysea) [reply src/js/components/Button/stories/CustomThemed/Active.js:144]: Thanks for catching, this story was staged with the brand refresh theme and not intended to stay when we merge. The line-height question can be handled in a separate PR if necessary. Reverting the story to the original.

File: src/js/components/Button/StyledButtonKind.js (hunk 1 above)
Comment (taysea) [reply src/js/components/Button/StyledButtonKind.js:47]: I'm inclined to leave as is.

**Comment by ericsoderberghp** [comment]
Chromatic picked up some things that look a bit off.

Pagination large vertical alignment:
Screenshot 2023-02-01 at 5 17 03 PM

Button large icon and padding don't look right here:
Screenshot 2023-02-01 at 5 18 18 PM

Custom themed Button changed a bit unexpectedly:
Screenshot 2023-02-01 at 5 19 23 PM

Screenshot 2023-02-01 at 5 20 12 PM

File: src/js/components/Button/Button.js [hunk 3]
```diff
@@ -242,6 +242,10 @@ const Button = forwardRef(
       return undefined; // pre-default, no kind
     }, [kindArg, kindObj, primary, secondary, theme]);
 
+    // for backwards compatibility, no-kind button theme did not
```
Comment (halocline) [inline src/js/components/Button/Button.js:245]: 👍

**Review by halocline — APPROVED** [review]

**Review by ericsoderberghp — APPROVED** [review]

**Pull request merged**